discrete_optimization/generic_tools/hyperparameters/hyperparameter.py

```diff
@@ -100,6 +100,12 @@ class IntegerHyperparameter(Hyperparameter):
 
     If None, the hyperparameter value has no upper bound.
 
+    """
+    step: Optional[int] = 1
+    """step to discretize.
+
+    If None, the step will be one.
+
     """
```

Comment:
Write rather
```
step: int = 1
"""Step to discretize."""
```

This is sufficient to have step default value being 1, and enforce an integer value.
Actually step=None would make trial.suggest_int() fail.